### PR TITLE
sdl_did_not_read_correctly_value_of_hmicapabilities_from_hmicapabilitiescachefile

### PR DESCRIPTION
**Pull request by RomanReznichenkoLuxoft**
Fixes #[12562](https://luxproject.luxoft.com/jira/browse/FORDTCN-12562)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
SDL sends RegisterAppInterface response with the incorrect value of "hmiCapabilities" parameters if HMICapabilitiesCacheFile exists. At the first start, we send the values of navigation and phoneCall from the HMI response from the hmicapabilities field. And the second time we had data in the HMICapabilitiesCacheFile file, we took the value from systemcapabilities, not from hmicapabilities

### CLA
- [ ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 1]
```diff
@@ -512,9 +512,6 @@ TEST_F(
   EXPECT_TRUE(navigation_capability_so.keyExists("getWayPointsEnabled"));
   EXPECT_TRUE(navigation_capability_so["sendLocationEnabled"].asBool());
   EXPECT_TRUE(navigation_capability_so["getWayPointsEnabled"].asBool());
-
-  // Since we have navigation capabilities, the feature should be supported
-  EXPECT_TRUE(hmi_capabilities_->navigation_supported());
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/hmi_capabilities_test.cc:517]: @RomanReznichenkoLuxoft why did you remove that expectation?

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 2]
```diff
@@ -524,8 +521,37 @@ TEST_F(HMICapabilitiesTest,
...
+  EXPECT_TRUE(ui_hmi_capability_so.keyExists("navigation"));
+  EXPECT_TRUE(ui_hmi_capability_so.keyExists("phoneCall"));
+  EXPECT_TRUE(ui_hmi_capability_so.keyExists("videoStreaming"));
+
+  hmi_capabilities_->set_phone_call_supported(
+      ui_hmi_capability_so["phoneCall"].asBool());
+  hmi_capabilities_->set_navigation_supported(
+      ui_hmi_capability_so["navigation"].asBool());
+  hmi_capabilities_->set_navigation_supported(
+      ui_hmi_capability_so["videoStreaming"].asBool());
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/hmi_capabilities_test.cc:550]: @RomanReznichenkoLuxoft why are you changing these fields? I believe the same code will be executed in `Init()` while parsing UI capabilities object

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 3]
```diff
@@ -840,8 +864,6 @@ TEST_F(HMICapabilitiesTest,
   hmi_capabilities_->Init(last_state_wrapper_);
 
   // Check system capabilities; only phone capability is available
-  EXPECT_FALSE(hmi_capabilities_->navigation_supported());
-  EXPECT_TRUE(hmi_capabilities_->phone_call_supported());
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/hmi_capabilities_test.cc:844]: @RomanReznichenkoLuxoft why did you removed that?

**Review by VadymLuchko — APPROVED** [review]

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 2 above)
Comment (RomanReznichenkoLuxoft) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: done

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 3 above)
Comment (RomanReznichenkoLuxoft) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: done

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 1 above)
Comment (RomanReznichenkoLuxoft) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: done

**Review by AKalinich-Luxoft — APPROVED** [review]

**Review by ychernysheva — APPROVED** [review]

**Review by OlhaVorobiova — APPROVED** [review]

**Comment by LitvinenkoIra** [comment]
moved to smartDeviceLink `https://github.com/smartdevicelink/sdl_core/pull/3820`
